src/Evolution/DiscontinuousGalerkin/DgElementArray.hpp

```diff
@@ -128,75 +128,17 @@ void DgElementArray<Metavariables, PhaseDepActionList>::allocate_array(
 
   const auto& blocks = domain.blocks();
 
-  // Only need the element distribution if the element weight has a value
-  // because then we have to use the space filling curve and not just use round
-  // robin.
-  domain::BlockZCurveProcDistribution<volume_dim> element_distribution{};
-  if (element_weight.has_value()) {
-    const std::unordered_map<ElementId<volume_dim>, double> element_costs =
-        domain::get_element_costs(blocks, initial_refinement_levels,
-                                  initial_extents, element_weight.value(),
-                                  quadrature);
-    element_distribution = domain::BlockZCurveProcDistribution<volume_dim>{
-        element_costs,   num_of_procs_to_use, blocks, initial_refinement_levels,
-        initial_extents, procs_to_ignore};
-  }
-
-  // Will be used to print domain diagnostic info
-  std::vector<size_t> elements_per_core(number_of_procs, 0_st);
-  std::vector<size_t> elements_per_node(number_of_nodes, 0_st);
-  std::vector<size_t> grid_points_per_core(number_of_procs, 0_st);
-  std::vector<size_t> grid_points_per_node(number_of_nodes, 0_st);
-
-  size_t which_proc = 0;
-  for (const auto& block : blocks) {
-    const auto& initial_ref_levs = initial_refinement_levels[block.id()];
-    const size_t grid_points_per_element = alg::accumulate(
-        initial_extents[block.id()], 1_st, std::multiplies<size_t>());
-
-    const std::vector<ElementId<volume_dim>> element_ids =
-        initial_element_ids(block.id(), initial_ref_levs);
-
-    // Value means ZCurve. nullopt means round robin
-    if (element_weight.has_value()) {
-      for (const auto& element_id : element_ids) {
-        const size_t target_proc =
-            element_distribution.get_proc_for_element(element_id);
+  Parallel::create_elements_using_distribution(
+      [&dg_element_array, &global_cache, &initialization_items](
+          const ElementId<volume_dim>& element_id, const size_t target_proc,
+          const size_t /*target_node*/) {
         dg_element_array(element_id)
             .insert(global_cache, initialization_items, target_proc);
+      },
+      element_weight, blocks, initial_extents, initial_refinement_levels,
+      quadrature,
 
```

Comment:
Is formatting really ok with this space in the function call?

Reply:
I'm not exactly sure what you mean by "okay". Clang format doesn't remove single blank lines (it does combine multiple blank lines into one!) and I found it easier to read the function call separating the slew of `size_t`s that represent hardware things from the parts that are tied to the element info :shrug:

Reply:
Hmm do we have a convention for this? I think there are many other instances of function calls that have a bajillion parameters that are difficult to read. Some have comments between lines to make it easier, some don't. Not gonna hold up this PR though, but we should probably figure out what to do